pandas/io/formats/style.py

```diff
@@ -1820,6 +1803,27 @@ def pipe(self, func: Callable, *args, **kwargs):
         return com.pipe(self, func, *args, **kwargs)
 
 
+class _Builtins:
+    @staticmethod
+    def _highlight_func(
+        data: FrameOrSeries,
+        props: str = "background-color: yellow;",
+        highlight: str = "max",
+        **kwargs,
+    ) -> np.ndarray:
+        """
+        Highlight the value in a Series or DataFrame by func with css-properties
+        """
+        if highlight == "max":
```

Comment:
this is extra complicated. just inline the conditionals in the functions themselvs.